File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        int left = 0 , right = n-1 , maxarea = 0  ;

        while(left < right){    
            int area = (right - left) * Math.min(height[left] , height[right]);
            maxarea = Math.max(area , maxarea);
            
            if(height[left] < height[right]){
               left ++;
            }
            else{
                right --;
            }

        }

        return maxarea;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        n = len(height)
        l = 0
        r = n - 1
        maxw = 0

        while l < r :
            if height[l] > height[r]:
                area = height[r]*(r-l)
                maxw = max(maxw , area)
                r -= 1
            elif height[l] <= height[r]:
                area = height[l]*(r-l)
                maxw = max(maxw , area)
                l += 1

        return maxw         


        